### PR TITLE
Completes assignment 4.1

### PR DESCRIPTION
**Pull request by avcopan**

### CODE REVIEW

File: assignment/scripts/function-practice.js [hunk 1]
```diff
@@ -14,31 +14,35 @@ console.log('Test - should say "Hello World!"', hello());
...
 // 2. Function to return an personalized hello, using the `name` argument.
 //    for example 'Hello, Jo!', or 'Hello, Stacy!'
-function helloName( name ) {
-  return;
+function helloName(name = 'my friend') {
+  return `Hello, ${name}!`;
 }
 // Remember to call the function to test
+console.log('Calling helloName("Jo")', helloName("Jo"));
+console.log('Calling helloName()', helloName());
```
Comment (delventhalz) [inline assignment/scripts/function-practice.js:22]: Nice use of a default argument.

File: assignment/scripts/function-practice.js [hunk 2]
```diff
@@ -14,31 +14,35 @@ console.log('Test - should say "Hello World!"', hello());
...
-
+console.log('multiplyThree(2, 3, 4)', multiplyThree(2, 3, 4));
 
 // 5. Function that will return true if a number is positive, 
 //    or greater than zero, and false otherwise
-function isPositive( number ) {
-  if ( number > 0 ){
-    return;
+function isPositive(number) {
+  if (number > 0) {
```
Comment (delventhalz) [inline assignment/scripts/function-practice.js:42]: Since `>` produces a boolean, you could also just return the condition directly:

```js
function isPositive(number) {
  return number > 0;
}
```

File: assignment/scripts/function-practice.js [hunk 3]
```diff
@@ -49,41 +53,107 @@ console.log( 'isPositive - should say false', isPositive(-3) );
 
 // 6. Function to return the _last_ item in an array. If the 
 //    array is empty, return `undefined`.
-function getLast( array ) {
-
+function getLast(array) {
+  let lastItem = undefined;
+  if (array.length) {
+    lastItem = array[array.length - 1];
```
Comment (delventhalz) [inline assignment/scripts/function-practice.js:59]: Since the default value for an array lookup is `undefined`, this function can also be simplified:

```js
function getLast(array) {
  return array[array.length - 1];
}
```

File: assignment/scripts/function-practice.js [hunk 4]
```diff
@@ -49,41 +53,107 @@ console.log( 'isPositive - should say false', isPositive(-3) );
...
+      positiveArray.push(element);
+    }
+  }
+  return positiveArray;
+}
+console.log(
+  'getPositiveElements([1, -2, 3, -4, -4, 5])',
+  getPositiveElements([1, -2, 3, -4, -4, 5])
+);
+console.log('getPositiveElements([])', getPositiveElements([]));
```
Comment (delventhalz) [inline assignment/scripts/function-practice.js:118]: Love all the edge case testing.

File: assignment/scripts/function-practice.js [hunk 5]
```diff
@@ -49,41 +53,107 @@ console.log( 'isPositive - should say false', isPositive(-3) );
...
+ * Use a RegEx to test whether a string is empty.
+ * Link: https://edabit.com/challenge/bAqxpvYmDuuvz4LQz */
+
+// Here's my function for checking if a string is empty.
+function emptyString(string, allowSpace=false) {
+  // This regExp is used if allowSpace is false. It only matches completely
+  // empty strings. It only matches when the string end is immediately followed
+  // by the string start.
+  let regExp = /^$/;
+  if (allowSpace) {
```
Comment (delventhalz) [inline assignment/scripts/function-practice.js:135]: This would have been a good spot for a ternary:

```js
let regExp = allowSpace ? /^[ \n\t]*$/ : /^$/;
```

File: assignment/scripts/function-practice.js [hunk 6]
```diff
@@ -49,41 +53,107 @@ console.log( 'isPositive - should say false', isPositive(-3) );
...
-  
+function find(value, array) {
+  for (let element of array) {
+    if (element === value) {
+      return true;
+    }
+  }
+  return false;
 }
+console.log("find('z', 'abcdefg')", find('z', 'abcdefg'))
```
Comment (delventhalz) [inline assignment/scripts/function-practice.js:77]: Is this function for an array or a string? Your implementation happens to work for both, but the documentation seems to indicate it is intended for arrays. Should probably use arrays to test it in that case.

**Review by delventhalz — COMMENTED** [review]
Great work. Love the "allowSpace" option.
